### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by gabrielcoder247**
### Form
In this milestone.
I added the following.

> Forms  comment and post forms

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,9 +1,21 @@
 class PostsController < ApplicationController
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:1]: - [ ] New and Create actions are not defined in your _posts controller_. Kindly ensure that the _posts controller_ have _new_ and _create_ actions similar to the one shown below:

```ruby
   def new
      @post = Post.new
    end

    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end

    # remind students about the strong parameters in this lesson https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end

	  ...

  end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class CommentsController < ApplicationController
+  def index; end
+
+  def show; end
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:4]: - [ ] New and Create actions are not defined in your _comments controller_. Kindly ensure that the _comments controller_ have _new_ and _create_ actions similar to the one shown below:

```ruby
...

  def new
    @comment = Comment.new
  end

	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end

  # remind students about the strong parameters in this lesson https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end

	...
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gabriel, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @gabrielcoder247 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+  <ul>
+    <% @posts.each do |post| %>
+    <%= link_to "#{root_path}users/#{@user.id}/posts/#{post.id}", class: "text-decoration-none" do %>
+      <li class="padding-20-px background-color-aliceblue">
+        <h3> <%= post.title %> </h3>
+        <p class="paragraph-text display-flex flex-direction-column">
+          <span><%= post.text %></span>
+          <span class="align-self-end">Comments: <%= @comments.where(post_id: post.id).count %>, Likes: <%= @likes.where(posts_id: post.id).count %></span>
+        </p>
+        <h4>Comments:</h4>
```
Comment (andersonlebon) [inline app/views/posts/index.html.erb:20]: ## [OPTIONAL]

- [ ] I would really suggest you implement the Like feature

**Pull request merged**